django_airavata/apps/api/static/django_airavata_api/js/utils/FetchUtils.js

```diff
@@ -28,20 +56,22 @@ export default {
         }
         return headers;
     },
-    post: function (url, body,queryParams = "", mediaType = "application/json") {
+    post: function (url, body, queryParams = "", mediaType = "application/json") {
         var headers = this.createHeaders(mediaType)
         // Browsers automatically handle content type for FormData request bodies
         if (body instanceof FormData) {
             headers.delete("Content-Type");
         }
-        console.log("post body",body);
-        url=parseQueryParams(url,queryParams)
+        console.log("post body", body);
+        url = parseQueryParams(url, queryParams);
+        incrementCount();
         return fetch(url, {
             method: 'post',
-            body: body,
+            body: JSON.stringify(body),
```

Comment:
Let's not stringify the body in the post (and put). This will break file uploads where the body is a FormData instance.  Or, what we could do is check to see if the body isn't a string instance and it's not a FormData instance then go ahead and stringify it.

Reply:
@eldhomathulla I've gone ahead and made this change: https://github.com/apache/airavata-django-portal/commit/18c7dc1139820e52690a41f88304f686f7c57b55